src/Microsoft.AspNet.SignalR.SqlServer/DbOperation.cs

```diff
@@ -180,43 +162,27 @@ private void LoggerCommand(DbCommand command)
         [SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times", Justification = "Disposed in async Finally block"),
          SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope", Justification = "Disposed in async Finally block")]
 #if ASPNET50
-        private void Execute<T>(Func<IDbCommand, Task<T>> commandFunc, TaskCompletionSource<T> tcs)
+        private async Task<T> Execute<T>(Func<IDbCommand, Task<T>> commandFunc)
 #else
-        private void Execute<T>(Func<DbCommand, Task<T>> commandFunc, TaskCompletionSource<T> tcs)
+        private async Task<T> Execute<T>(Func<DbCommand, Task<T>> commandFunc)
 #endif
         {
-#if ASPNET50
-            IDbConnection connection = null;
-#else
-            DbConnection connection = null;
-#endif
-            try
+            using (var connection = _dbProviderFactory.CreateConnection())
             {
-                connection = _dbProviderFactory.CreateConnection();
                 connection.ConnectionString = ConnectionString;
                 var command = CreateCommand(connection);
 
                 connection.Open();
 
-                commandFunc(command)
-                    .Then(result => tcs.SetResult(result))
-                    .Catch(ex => tcs.SetUnwrappedException(ex), Logger)
-                    .Finally(state =>
-                    {
-                        var conn = (DbConnection)state;
-                        if (conn != null)
-                        {
-                            conn.Dispose();
-                        }
-                    }, connection);
-            }
-            catch (Exception)
-            {
-                if (connection != null)
+                try
+                {
```

Comment:
I think we should call `LoggerCommand` here like we do in the other `Execute` method.